### ISSUE TITLE
Supervisor 2022.06.x results in unusable system on older os releases

### ISSUE THREAD

**Issue by mgolisch** [home-assistant/operating-system#2005]
### Describe the issue you are experiencing

I believe the problem is the same as https://github.com/home-assistant/supervisor/issues/3692.
There were a number of people in the discord that ran into that issue.
Not sure if there is anything that can be done about that.

Seems all those people were on realy old os releases betwen 4.x and 5.x

Also not sure how they even ended up with that new supervisor, they all said they didnt update it themselves.

Also for one of them they seemed to run into a problem where hasos-supervisor wasnt able to download the supervisor image at all due to ratelimiting.
Presumably the new supervisor failed and then it just tried repulling the latest release untill it ran into the ratelimiting?
Is there a reason /usr/sbin/hasos-supervisor does not bydefault pull the image from ghcr.io?

### What operating system image do you use?

ova (for Virtual Machines)

### What version of Home Assistant Operating System is installed?

4.x -?

### Did you upgrade the Operating System.

No

### Steps to reproduce the issue

1. have old OS realse
2. Update Supervisor to 2022.06.x
3. Everything is broken
4. Cant update

### Anything in the Supervisor logs that might be useful for us?

```txt
They all had the same keyerror cgroup error as in the linked bug report.
```

### Anything in the Host logs that might be useful for us?

```txt
-
```

### System Health information

-

### Additional information

-

**Comment by agners** [comment]
> Seems all those people were on realy old os releases betwen 4.x and 5.x

Those are unsupported Home Assistant OS versions at this point, see https://github.com/home-assistant/architecture/blob/master/adr/0015-home-assistant-os.md#supported-operating-systems-and-versions.

While we try to not intentionally break old versions, at one point it usually happens. It seems that is the case here.

> Also not sure how they even ended up with that new supervisor, they all said they didnt update it themselves.

The Supervisor updates automatically without any user interaction. This is by design. But there is a feature request which would allow to block automatic updates (see https://community.home-assistant.io/t/feature-request-block-supervisor-auto-updates/112743).

> Also for one of them they seemed to run into a problem where hasos-supervisor wasnt able to download the supervisor image at all due to ratelimiting.
> Presumably the new supervisor failed and then it just tried repulling the latest release untill it ran into the ratelimiting?
Is there a reason /usr/sbin/hasos-supervisor does not bydefault pull the image from ghcr.io?

There have been reasons in the past why we continued using Docker Hub registry, but now ghcr should be fine. I created a PR already (#2005). It probably also make sense to lower the rate limit (#2010).

**Comment by agners** [comment]
It wasn't intentionally. Supervisor updates don't get tested on unsupported systems, so that explains why it made it to the stable channel without us noticing.

It has been fixed meanwhile (in `2022.07.0`), and if you just leave the system running, it will update to the fixed version at one point.

That said, if you are using such old HAOS, you should update your operating system :wink:

**Comment by mykola-dev** [comment]
thanks for clarification.
i was using raspbian (debian stretch) on my rpi. when it just works i prefer to not update.
but when it suddenly broke, i upgraded to the debian buster, upgraded docker and my home automation is back to life.
